packages/ckeditor5-engine/src/model/utils/insertcontent.ts

```diff
@@ -85,21 +85,28 @@ export default function insertContent(
 
 		if ( content.is( 'documentFragment' ) && content.markers ) {
 			for ( const [ name, range ] of content.markers ) {
+				const { start, end } = range;
+
 				markersPosition.push( {
-					position: range.start,
+					position: start,
 					name
 				}, {
-					position: range.end,
+					position: end,
 					name
 				} );
 			}
 
 			markersPosition.sort( ( { position: posA }, { position: posB } ) => posA.isBefore( posB ) ? 1 : -1 );
 
 			for ( const { position, name } of markersPosition ) {
-				const fakeElement = writer.createElement( '$marker', { 'data-name': name } );
+				let fakeElement = null;
+				const isBeginning = position.path.length === 1 && position.path[ 0 ] === 0;
+				const isEnding = position.path.length === 1 && position.path[ 0 ] === content.childCount;
```

Comment:
`content.maxOffset`. But you should also be able to use `Position#isAtEnd()` 🤔.

Reply:
I think I'd go with `position.parent === content && position.isAtStart` and `position.isAtEnd`. Will be more clear what is going on here.